lib/r10k/util/subprocess/runner/childprocess.rb

```diff
@@ -0,0 +1,45 @@
+require 'r10k/util/subprocess/runner'
+require 'r10k/util/subprocess/runner/pump'
+require 'childprocess'
+
+class R10K::Util::Subprocess::Runner::Childprocess < R10K::Util::Subprocess::Runner
+
+  def initialize(argv)
+    @argv = argv
+  end
+
+  def run
+    process = ChildProcess.build(*@argv)
+    process.cwd = @cwd if @cwd
+
+    stdout = ''
+    stderr = ''
+    exit_code = 254
+
+    stdout_r, stdout_w = IO.pipe
+    stdout_pump = R10K::Util::Subprocess::Runner::Pump.new(stdout_r)
+    stderr_r, stderr_w = IO.pipe
+    stderr_pump = R10K::Util::Subprocess::Runner::Pump.new(stderr_r)
+
+    process.io.stdout = stdout_w
+    process.io.stderr = stderr_w
+
+    begin
+      stdout_pump.start
+      stderr_pump.start
+      process.start
+      process.wait
+      stdout_pump.halt!
+      stderr_pump.halt!
```

Comment:
Fixed in https://github.com/adrienthebo/r10k/commit/d816ea9cbf53806477cfc718f20ea1bd54960f07.